cmd/ghemoji/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/donatj/ghemoji"
+)
+
+func main() {
+	// right now, we are not using flag, but we could
+	// so let's plan it
+	flag.Usage = usage
+
+	input, err := readInput()
+	if err != nil {
+		flag.Usage()
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(2)
+	}
+
+	if input == "" {
+		flag.Usage()
+		fmt.Fprintln(os.Stderr, "no input provided")
+		os.Exit(2)
+	}
+
+	fmt.Fprintln(os.Stdout, ghemoji.ReplaceAll(input))
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage: %s [text]\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "or pipe text to the program\n")
+	flag.PrintDefaults()
+}
+
+// readInput reads input from stdin if stdin is a named pipe.
+// Otherwise, it joins the command-line arguments with a space separator.
+func readInput() (string, error) {
+	info, err := os.Stdin.Stat()
+	if err != nil {
+		return "", fmt.Errorf("getting stat from stdin: %w", err)
+	}
+
+	if info.Mode()&os.ModeNamedPipe != 0 {
```

Comment:
How’s about we use https://github.com/mattn/go-isatty ? It’s cross country (read: Works in Window 😅)

Reply:
Thinking about this more, I might be inclined just to reverse the logic - and do it like this?

```go
func readInput() (string, error) {
	text := strings.Join(flag.Args(), " ")
	if text != "" {
		return text, nil
	}

	input, err := io.ReadAll(os.Stdin)
	if err != nil {
		return "", fmt.Errorf("reading from stdin: %wv", err)
	}
	return string(input), nil
}
```

It would give it a more similar behaviour to `cat` I think which is more … unixy

Reply:
> How’s about we use https://github.com/mattn/go-isatty ? It’s cross platform (read: Works in Window 😅)

the code is not about checking if the code is a terminal (what isatty does) it's about reading the content of Stdin only when piped.

Reply:
> Thinking about this more, I might be inclined just to instead reverse the logic - and do it like this?
> 
> ```go
> func readInput() (string, error) {
> 	text := strings.Join(flag.Args(), " ")
> 	if text != "" {
> 		return text, nil
> 	}
> 
> 	input, err := io.ReadAll(os.Stdin)
> 	if err != nil {
> 		return "", fmt.Errorf("reading from stdin: %w", err)
> 	}
> 	return string(input), nil
> }
> ```
> 
> It would give it a more similar behaviour to `cat` I think which is more … unixy

I will have a look, I'm unsure